packages/core/src/internal/execution/execution-engine.ts

```diff
@@ -74,6 +74,11 @@ export class ExecutionEngine {
       defaultSender
     );
 
+    await this._executionStrategy.init(
```

Comment:
@zoeyTM you initially had this moved up into the `deploy` function; I moved it back down to the execution engine. My reasoning is that we don't want to do an init on a strategy until we are sure that something is definitely going to be executed - so it should have passed all previous validations/checks/hurdles.
Can you recall the initial reason for moving it up to the `deploy` function?

Reply:
I don't think there was any important reason. I just needed the deployment loader and it was the first place I was able to pass it in. This change looks good to me.